Hierarchical3DRegistration/Hierarchical3DRegistration.py

```diff
@@ -504,7 +504,7 @@ def registerSequence(
 
         try:
             self.isRunning = True
-            for idx in range(startFrame, endFrame):
+            for idx in range(startFrame, endFrame + 1):
```

Comment:
> while it is being iterated over does not seem to be correct.

To follow-up offline discussion with @sbelsk, since this code was known to work, this will likely be reviewed/addressed in follow-up pull requests.

Reply:
> Considering `endFrame` is set to the value of `CTSelectorNode.GetNumberOfDataNodes()`, using `range(startFrame, endFrame)` seems to be correct.
> 
> Indeed, considering the items in a sequence are retrieved starting with 0, calling `self.autoscoperLogic.getItemInSequence(ctSequence, idx)` with idx set to the number of item would not be correct.

Makes sense, thanks for pointing out the inconsistency. The UI frame selector slider was set to go from minimum=`0` to up to maximum=`CTSelectorNode.GetNumberOfDataNodes()` which would add one more frame than there really is. 

I pushed a fix to make sure the maximum value the user is able to select is actually the last frame in the sequence. I believe all other places in the code use `CTSelectorNode.GetNumberOfDataNodes()` consistently such that the first frame is treated as index `0` and the last is treated as `GetNumberOfDataNodes()-1`. (E.g., error checks in [`TreeNode.py`](https://github.com/sbelsk/SlicerAutoscoperM/blob/fix-off-by-one-registration-end-frame/Hierarchical3DRegistration/Hierarchical3DRegistrationLib/TreeNode.py) of the form `if idx >= self.transformSequence.GetNumberOfDataNodes():` and the loop that goes `for idx in range(self.transformSequence.GetNumberOfDataNodes()):`).